vendor-bin/box/composer.json

```diff
@@ -0,0 +1,6 @@
+{
+    "require": {
+        "humbug/box": "^3.0@dev",
```

Comment:
`dev` means it's fine with any update, and that any update may bring BC breaks, as it's not yet stable.
why not 3.0.0-alpha.4 ?

Reply:
Because `alpha.3` was released just yesterday :) It's true but that's also why I committed the `.lock`: you use a fixed version that works and you can try to update at any given time. Since it's alpha, there is potential BC breaks (I try to keep them minimal obviously) so there is not much difference in practice

Reply:
you try to convince me I can use `symfony/symfony: *` instead of `symfony/symfony: ^4@stable` because i have `.lock` file ?
well... i'm not convinced.

I want to be able to run `composer upgrade` for getting non-BC breaks always. having `*` or `@dev` on not-yet-stable package will make that impossible.

Reply:
Yes you _could_ do it. I'm not suggesting to apply my comment to everything neither: as you said you want to have stable and non-BC breaks so using the right constraint for that helps.

However in the case of Box it is unstable, `@dev` or `@alpha`, so it's just a matter of how you manage that instability. Since you try really hard to not rely on a `.lock` file, would fixing it at a specific version work better here instead of a ranged constraint?

Reply:
If using that approach, i'm cutting off possibility to run `composer upgrade` and receive no BC breaks from performing such action. Not nice for me.

This case here is exception - because you proposed to use not-yet-stable package.
Ultimatelly, we would just put `^3` (= `^3.0@stable`)
As I believe, in `.json` one shall have as wide range of supported versions as possible (, and if he needs `.lock` file, then locking down to concrete SHA).
Here, it would be concrete alpha, until stable version would be released.